Propose some contribution guidelines. (#185)
statechannels/go-nitro: contributing.md

# Contributing

Contributions to go-nitro are welcome from anyone.

> Magmo (at Consensys Mesh) are developing go-nitro under contract for the Filecoin Foundation. As a consequence, we are obligated to maintain a certain level of quality.

Reviewer: ```suggestion
Magmo (at Consensys Mesh) are developing go-nitro under contract for the Filecoin Foundation. As a consequence, we reserve discretionary veto rights on feature additions, architectural decisions, etc., which may hinder progress toward the specific aims of this contract.
```

## Contribution lifecycle

Magmo uses [Github Projects](https://github.com/statechannels/go-nitro/projects/1) for go-nitro project management. We will aim to keep a handful of "good first issues" in [the pipeline](https://github.com/statechannels/go-nitro/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22), which are a good place to start.

> Before spending lots of time working on an issue, consider asking us for feedback. We would love to help make your contributions more successful!

Reviewer: Consider:
```suggestion
Before spending lots of time working on an issue, consider asking us for feedback via [Slack](https://statechannels.slack.com/archives/C02J81JFD3J), the [issue tracker](https://github.com/statechannels/go-nitro/issues), or [email](magmo@mesh.xyz). We would love to help make your contributions more successful!
```

caveat: I'm not sure whether that email link will render as a `mailto`.

Pull requests (internal or external) will be reviewed by one or two members of Magmo. We aim to review most contributions within one business day.

We include a PR template to serve as a **guideline**, not a **rule**, to communicate the code culture we wish to maintain in this repository.

## Style

> As of now, Magmo is new to the go language, and has no specific opinions about go style. As a rule of thumb, we defer to the [Effective Go](https://go.dev/doc/effective_go) document [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences) as "style guides".

Author: ```suggestion
As of now, Magmo is new to the go language, and has no specific opinions about go style. As a rule of thumb, we defer to the [Effective Go](https://go.dev/doc/effective_go) document [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) as "style guides".
```

> ## Judgement

Reviewer 2: Done in a229754
